statistics/shortest_time_to_get_all_singles_and_averages.rb

```diff
@@ -15,39 +19,34 @@ def query
         start_date,
         best,
         average
-      FROM Results
-      JOIN Persons person ON person.id = personId and person.subId = 1
+      FROM (
+        -- People who have single for every official event.
+        SELECT personId FROM RanksSingle GROUP BY personId HAVING COUNT(eventId) = #{Events::OFFICIAL.length}
+      ) AS all_events_people
+      JOIN (
+        -- People who have average for every official event.
+        SELECT personId FROM RanksAverage GROUP BY personId HAVING COUNT(eventId) = #{NUM_EVENTS_WITH_AVERAGES}
```

Comment:
Same as before, just additionally we ensure the people have average for every event as well.